Lambdas/GHHooks/SecretsRetriever.swift

```diff
@@ -23,6 +41,30 @@ struct SecretsRetriever {
         guard let secret = secret.secretString else {
             throw Errors.secretNotFound(arn: arn)
         }
-        return secret
+        return Secret(secret)
+    }
+
+    private func withLock<T>(block: () async throws -> T) async rethrows -> T {
+        if isLoading {
```

Comment:
To no one's surprise more than my own, I think one of my original suggestions here was actually still correct:
```suggestion
        while isLoading {
```
As per https://www.wwdcnotes.com/notes/wwdc21/10133/, when you cross an `await` in an `actor`, you must re-verify mutable state.